### DISCUSSION TITLE
How can I implement the TPC-DS Load Test of Duckdb

### DISCUSSION THREAD

**Discussion by woshimajintao** [duckdb/duckdb#9421] [Q&A] [answered]
How can I implement the TPC-DS Load Test of Duckdb?
Is there anyone give me some suggestions?

**Comment by Tishj** [comment]
We actually have a TPC-DS extension 👍 
```sql
CALL dsdgen(sf=1)
PRAGMA tpcds(1); -- replace 1 with the query number you want to run
```

**Reply by woshimajintao** [reply]
Comparison of PRAGMA runtime and not using PRAGMA runtime when completing the tpc-ds test using Duckdb:

We try to use time module to complete the statistics on the execution time of the PRAGMA
query.The perf counter() function always returns the float value of time in seconds. 
import time
s t a r t = time . perf_counter ( )
%s q l PRAGMA tpcds ( 14 ) ;
end = time . perf_counter ( )
p r i n t ( ’ execution time i s { } s ’ . format ( end− s t a r t ) )
execution time i s 0 .9024780000000874 s

But if we didn’t use the PRAGMA extension, but instead implement the query statement directly
into duckdb.sql() for the time statistics, we found that this execution ime is cut down dramatically,
even by a factor of hundred.
execution time i s 0 .004080400000020745 s

Could you explain why  this is happening.I think the PRAGMA is just like a black box but i dont know how you finish it.So i am very confused about it!

**Reply by l1t1** [reply]
try run it in CLI
```
.timer on
PRAGMA tpcds(1);
```

**Reply by woshimajintao** [reply]
I want to get the every query time from PRAGMA tpcds(1) to PRAGMA tpcds(99) by a loop in CLI.But I dont know how to implement it.Could you please give me some suggestions?

**Reply by l1t1** [reply, marked as answer]
.mode c
.header off
.output a.sql
select ' PRAGMA tpch('||i||');' from range(22)t(i);
.output
.read a.sql

**Reply by woshimajintao** [reply]
Many thanks!I modified the 'TPCH' to 'TPCDS'.And I wonder how to add sum the every query time together in the CLI.

**Reply by woshimajintao** [reply]
I would like to print the every query time into a bar chart.

**Comment by l1t1** [comment]
I test tpch q1, no different beween sql and pragma

https://github.com/duckdb/duckdb/blob/main/extension/tpch/dbgen/queries/q01.sql

[tpchq1.txt](https://github.com/duckdb/duckdb/files/13184513/tpchq1.txt)

**Reply by woshimajintao** [reply]
Yeah!Thanks!I had tried it.It is true!

**Comment by l1t1** [comment]
```sql
.mode c
.header off
.output a1.sql
select 'load tpch;';
select 'call dbgen(sf=0.1);';
select '.timer on';
select ' PRAGMA tpch('||i||');' from range(1,23)t(i);
.output
.exit
```
then run 
./duckdb <a1.sql |grep real

**Comment by l1t1** [comment]
time ./duckdb <a1.sql |grep real 
to get total tine

**Comment by l1t1** [comment]
save following as /a2.sql
```
.mode c
.header off
select 'load tpch;';
select 'call dbgen(sf=0.1);';
select '.timer on';
select ' PRAGMA tpch('||i||');' from range(1,23)t(i);
```
run `duckdb091 </a2.sql |  duckdb091 |/timer64 find "real"` at cmd console, timer64 is a tool similar as linux time
